src/form/Form.jsx

```diff
@@ -577,63 +576,48 @@ function Form({
       // goes to Feathery origin, while the previous
       // request goes to our CDN
       clientInstance
-        .fetchSession()
-        .then(async (session) => {
-          setIntegrations(session.integrations);
-          const newSession = await initializeIntegrations(
-            session.integrations,
-            clientInstance
-          );
-          if (newSession) session = newSession;
-
-          fetchPromise
-            .then(
-              async ([
-                steps,
-                {
-                  save_user_location: saveUserLocation,
-                  save_user_data: saveUserData
-                }
-              ]) => {
-                const usePrevious =
-                  usePreviousUserData === null
-                    ? saveUserData
-                    : usePreviousUserData;
-                if (!usePrevious)
-                  // Pass initial values to overwrite values when form history is off
-                  clientInstance.setDefaultFormValues({
-                    steps: Object.values(steps),
-                    additionalValues: initialValues,
-                    override: true
-                  });
-                if (!isObjectEmpty(initialValues))
-                  clientInstance.submitCustom(initialValues);
-                const hashKey = decodeURI(location.hash.substr(1));
-                const newKey =
-                  initialStepId ||
-                  (hashKey && hashKey in steps && hashKey) ||
-                  (saveUserLocation && session.current_step_key) ||
-                  getOrigin(steps).key;
-                setFirstStep(newKey);
-                history.replace(
-                  location.pathname + location.search + `#${newKey}`
-                );
+        .fetchSession(fetchPromise)
```

Comment:
maybe rename `fetchPromise` to `formPromise`? would match the param name in the `fetchSession` fn and make it more clear which fetch the promise is for

Reply:
good point